config/config.go

```diff
@@ -346,6 +393,47 @@ func setYamlValue(node *yaml.Node, path []string, value *yaml.Node) []string {
 	return []string{}
 }
 
+func deleteYamlValue(node *yaml.Node, path []string) []string {
+
+	switch node.Kind {
+	case yaml.DocumentNode:
+		for _, c := range node.Content {
+			path = deleteYamlValue(c, path)
+		}
+
+	case yaml.MappingNode:
+		for i := 0; i < len(node.Content); i += 2 {
+			// Keys/Values are inline. Count by twos to get it right.
+			key := node.Content[i]
+			val := node.Content[i+1]
```

Comment:
what if the input is malformed and contains an odd number of elements? Would we index out of bounds in such a case?

Reply:
Good question, I copied the logic from `setYamlValue` above but will investigate further

Reply:
Even if I have no value (nil) in my yaml it seems to work

config:
```
global:
    conversion_parallelism:
```
then
```
earthly config global.conversion_parallelism --delete
```
runs cleanly. printed the values of `[i]` and `[i+1]` when we get to that spot in the yaml
```
key: &{Kind:8 Style:0 Tag:!!str Value:conversion_parallelism Anchor: Alias:<nil> Content:[] HeadComment: LineComment: FootComment: Line:2 Column:5}
val: &{Kind:8 Style:0 Tag:!!null Value: Anchor: Alias:<nil> Content:[] HeadComment: LineComment: FootComment: Line:2 Column:28}
```
and even if there's nothing there
```
key: &{Kind:8 Style:0 Tag:!!str Value:global Anchor: Alias:<nil> Content:[] HeadComment: LineComment: FootComment: Line:1 Column:1}
val: &{Kind:4 Style:32 Tag:!!map Value: Anchor: Alias:<nil> Content:[] HeadComment: LineComment: FootComment: Line:1 Column:9}
```
Seems to always have a key and val with at least nil within the yaml node's Content array.

What do you think @alexcb can you think of a way it might not be the case?

Reply:
that's good to know, it was just something that jumped out, but if our tests check for that case and they pass, they seem like the best source of truth :+1: